build/archives/BUILD.gn

```diff
@@ -20,14 +20,18 @@ generated_file("artifacts_entitlement_config") {
   }
 }
 
-if (build_engine_artifacts) {
+if (build_engine_artifacts && !is_android) {
   zip_bundle("artifacts") {
     deps = [
       "//flutter/flutter_frontend_server:frontend_server",
       "//flutter/impeller/compiler:impellerc",
+      "//flutter/impeller/compiler:impellerc_license",
       "//flutter/impeller/tessellator:tessellator_shared",
       "//flutter/lib/snapshot:generate_snapshot_bin",
       "//flutter/shell/testing:testing",
+      "//flutter/tools/path_ops:path_ops",
+      "//flutter/tools/path_ops:path_ops_license",
+      "//third_party/dart/runtime/bin:gen_snapshot",
```

Comment:
I think the existing build gets `gen_snapshot` for this from the target `//flutter/lib/snapshot` here https://github.com/flutter/engine/blob/main/lib/snapshot/BUILD.gn#L248. There seems to be some logic in that file for building `gen_snapshot` that goes beyond a simple dependency on `//third_party/dart/runtime/bin:gen_snapshot`, so I'm a little worried that some differences might creep in here.

Reply:
After changing to `//flutter/lib/snapshot` gn complains about `//flutter/lib/snapshot` not generating the gen_snapshot binary.

```
The file:
  //out/host_profile/gen_snapshot
is listed as an input or source for the target:
  //flutter/build/archives:artifacts
but this file was not generated by any dependencies of the target. The target
that generates the file is:
  //third_party/dart/runtime/bin:gen_snapshot
```

Reply:
I think the issue there is that the tool for the `compiled_action` isn't listed as a public dependency, but I took a closer look, and I don't think the logic from `//flutter/lib/snapshot` is needed here. That target needs special logic for producing a `gen_snapshot` that can run on a 64-bit intel host and produce a 32-bit intel snapshot, which we don't actually ship. So this PR lgtm.